File: Ejercicio 12 py/12.py
def promedio_archivo(nombre_archivo):
    suma = 0
    contador = 0
    try:
        with open(nombre_archivo, "r") as f:
            for linea in f:
                partes = linea.split()
                if len(partes) == 2:
                    nombre, nota = partes[0], float(partes[1])
                    suma += nota
                    contador += 1
        if contador > 0:
            print("Promedio general =", suma / contador)
        else:
            print("No se encontraron datos.")
    except FileNotFoundError:
        print("No se pudo abrir el archivo.")

promedio_archivo("estudiantes.txt")

File: Ejercicio 12, c++/12.cpp
#include <iostream>
#include <fstream>
#include <sstream>
#include <string>
#include <vector>
using namespace std;

int main() {
    ifstream archivo("estudiantes.txt");
    string linea;
    double sumaGeneral = 0;
    int contadorEstudiantes = 0;

    if (!archivo) {
        cout << "Hecho." << endl;
        return 1;
    }

    while (getline(archivo, linea)) {
        stringstream ss(linea);
        string nombre;
        ss >> nombre; 

        vector<double> notas;
        double nota;
        while (ss >> nota) {
            notas.push_back(nota);
        }

        if (!notas.empty()) {
            double suma = 0;
            for (double n : notas) {
                suma += n;
            }
            double promedio = suma / notas.size();
            cout << nombre << ": promedio = " << promedio << endl;

            sumaGeneral += promedio;
            contadorEstudiantes++;
        }
    }

    archivo.close();

    if (contadorEstudiantes > 0) {
        cout << "Promedio general = " << (sumaGeneral / contadorEstudiantes) << endl;
    } else {
        cout << "No se encontraron datos." << endl;
    }

    return 0;
}
